src/linker/Linker/MarkingHelpers.cs

```diff
@@ -11,12 +11,33 @@ public MarkingHelpers (LinkContext context)
 			_context = context;
 		}
 
-		public void MarkExportedType (ExportedType type, ModuleDefinition module, in DependencyInfo reason)
+		public void MarkMatchingExportedType (TypeDefinition typeToMatch, AssemblyDefinition assembly, in DependencyInfo reason)
```

Comment:
How is this working when there are multiple typeforwarders in the chain?

Reply:
All copy assemblies will go through the `MarkEntireAssembly` method, which marks all exported types and the forwarded scope of every type reference (in particular, this will mark the scope of exported types -- which is the assembly to which the exported type ``points to''); this is enough for keeping all necessary forwarders in a chain.

e.g., if we have the following chain:
```
[copy] A -> [copy] B -> [link] C -> [link] D -> [implementation] E
```

and a static typeref to `A`, the logic would mark `A` and `B` (since these are exported types in a copy assembly), `C` would be marked by `MarkForwardedScope(B)`, `D` would be trimmed and `E` would be marked since it's the type that `C` resolves to:

https://github.com/mono/linker/blob/ea3525008ab1496e304d90476e543565f2cc522e/src/linker/Linker.Steps/MarkStep.cs#L1344-L1350